code-challenges.test.js

```diff
@@ -11,51 +11,272 @@
 // Run the file with the following command: $ yarn jest
 
 
+
+
+// describe(" ", () => {
+//     it(" ", () => {
+//       expect( ).toEqual()
+//     })
+//   })
+
+
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+//I will name my funciton secretCoder
+
+describe("secretCoder", () => {
+    const secretCodeWord1 = "Lackadaisical"
+  // Expected output: "L4ck4d41s1c4l"
+  const secretCodeWord2 = "Gobbledygook"
+  // Expected output: "G0bbl3dyg00k"
+  const secretCodeWord3 = "Eccentric"
+  // Expected output: "3cc3ntr1c"
+    it("takes in a string and returns a coded message", () => {
+      expect(secretCoder(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(secretCoder(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(secretCoder(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+  })
 
+//ran test and failed -> function was not defined
 
 
 // b) Create the function that makes the test pass.
 
+//first I will create a function that takes in a string
+// it will take the string and change the listed letters above to the numbers listed
+//I will change the string into an array so that I can iterate over each letter 
+// I will also need to make sure all the letters are lowercase 
+// I will create an array that holds the string that is all lowercase
+// as I iterate over the array I will change the letters to the numbers
+//My funciton will then return a new array 
+//I will then change the array back into a string 
+//I attempted to refactor with ternary operators to cut down on the amount of code but didn't have much luck. 
+// I also tried to use only one .toLowerCase method to blanket all the strings but realized it doesn't work when returning the string (haha) so I put them in each conditional statement.
+
+const secretCoder = (string) =>{
+    let arrOne = string.split("")
+    arrTwo = arrOne.map((value, index) => {
+    if(value.toLowerCase() === "a"){
+        return 4
+    }
+    else if(value.toLowerCase() === "e"){
+        return 3
+    }
+        else if(value.toLowerCase() == "i"){
+            return 1
+        }  
+            else if(value.toLowerCase() == "o"){
+                return 0
+            }  
+            return value
+            
+            })
+            return arrTwo.join("")
+            }
+
```

Comment:
Your logic looks great. Keep an eye on your indentation. It might just be the GitHub UI, but it looks a little off to me.